synapse/handlers/message.py

```diff
@@ -901,6 +913,29 @@ def persist_and_notify_client_event(
                         "Room alias %s does not point to the room" % (room_alias_str,),
                     )
 
+            # Check that alt_aliases is the proper form (and that each alias is
+            # currently valid).
+            alt_aliases = event.content.get("alt_aliases", [])
+            if alt_aliases and not isinstance(alt_aliases, collections.Sequence):
```

Comment:
So...turns out str is a sequence (`isinstance('foo', collections.Sequence) == True`).

#6941 used `collections.Sequence` as well, so I might need to revise that. (That PR actually has some similar code to this anyway, so might need to think about that.)

Reply:
I added an additional commit (6225a6f3f3c9d417fcb504a0bac770da49a343ed) to handle this.